state/status.go

```diff
@@ -157,7 +157,39 @@ func (m *ModelStatus) UnitWorkload(unitName string) (status.StatusInfo, error) {
 		return info, nil
 	}
 
-	return m.getStatus(unitGlobalKey(unitName), "workload")
+	// (for CAAS models) Use cloud container status over unit if the cloud
```

Comment:
This needs to be tested; I think the tests for this stuff may sadly be in cmd/juju/status
(would need to check)